src/core/mount.c

```diff
@@ -440,6 +440,9 @@ static int mount_add_quota_dependencies(Mount *m) {
         if (!mount_needs_quota(p))
                 return 0;
 
+        if (!fstype_needs_quota(p->fstype));
```

Comment:
This should be called in `mount_needs_quota()`. At least this must be conditionalized with the existence of `p->fstype`.